### ISSUE TITLE
Got unknown type role=<PromptMessageRole.TOOL:'tool'>

### ISSUE THREAD

**Issue by drfenixlinghu** [langgenius/dify#5353]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.11

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

every time i use a tool in the agent mode, after the tool perform its job, the agent no longer accept any further prompts. seems the agent feels that it has used the tool and the tool has done the job, therefore no more action is required or accepted. This happens with all tools available to dify. 
![d1ed109e672c393659c6f3c2e928385](https://github.com/langgenius/dify/assets/156991093/f24c4ac8-b70d-405a-9f6e-65dd9962a8c6)
![0efff3d09b4986e4a6f215f96423143](https://github.com/langgenius/dify/assets/156991093/53437151-8d53-4951-86d2-30552866a5ca)

anyone has the same issue?

### ✔️ Expected Behavior

to be able to continue with the conversation

### ❌ Actual Behavior

cannot go on with the conversation.

**Comment by crazywoola** [comment]
I think this should be fixed in latest main branch.

**Comment by cjdxhjj** [comment]
the error exist a long time since 0.6.10 and can't resolved at 0.6.11

**Comment by crazywoola** [comment]
Link https://github.com/langgenius/dify/issues/5304

**Comment by crazywoola** [comment]
Yes we just fixed yesterday. https://github.com/langgenius/dify/pull/5245

**Comment by mmm8955405** [comment]
最新mast版本， sd遇到同样错误

**Comment by Yeuoly** [comment]
which model are you using for this?

**Comment by fandych** [comment]
@Yeuoly I'm using GLM-4 with the same problem.

**Comment by dyz1027543829** [comment]
Does this problem have been fixed until now? @crazywoola

**Comment by fandych** [comment]
> Does this problem have been fixed until now? @crazywoola

I think it can be closed. It was fixed from v0.6.13.
